app/src/main/java/com/example/fitjournal/journalEntry/domain/JournalEntryDetailsViewModel.kt

```diff
@@ -0,0 +1,15 @@
+package com.example.fitjournal.journalEntry.domain
+
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.ViewModelProvider
+
+class JournalEntryDetailsViewModelFactory(private val selectedWorkoutDetail: WorkoutDetail?) :
+    ViewModelProvider.NewInstanceFactory() {
+    override fun <T : ViewModel> create(modelClass: Class<T>): T =
+        JournalEntryDetailsViewModel(selectedWorkoutDetail) as T
+}
```

Comment:
idk if we need this VM for journal entry, we may be fine with one vm for journal entry and journal entry details

Reply:
I can merge them and probably that would allow me to remove the selectedWorkoutDetail you were asking earlier.
in iOS for each view we have its own ViewModel, it is not a common practice that different views share the same ViewModel, that is why I was going through all this hoops.

Reply:
Normally i also agree, but being the first view has vary little logic effort and is tied to the second view combining them shouldnt be an issue here. Would just need to clear vm data on navigate up